openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+            if self._sum_sq is not None:
+                self._sum_sq = np.reshape(self._sum_sq.toarray(), self.shape)
+            if self._mean is not None:
+                self._mean = np.reshape(self._mean.toarray(), self.shape)
+            if self._std_dev is not None:
+                self._std_dev = np.reshape(self._std_dev.toarray(), self.shape)
+            self._sparse = False
+
+    def remove_score(self, score):
+        """Remove a score from the tally
+
+        Parameters
+        ----------
+        score : str
+            Score to remove
+
+        """
+
+        if score not in self.scores:
+            msg = 'Unable to remove score "{0}" from Tally ID="{1}" since ' \
+                  'the Tally does not contain this score'.format(score, self.id)
+            ValueError(msg)
+
+        self._scores.remove(score)
+
+    def remove_filter(self, old_filter):
+        """Remove a filter from the tally
+
+        Parameters
+        ----------
+        old_filter : openmc.Filter
+            Filter to remove
+
+        """
+
+        if old_filter not in self.filters:
+            msg = 'Unable to remove filter "{0}" from Tally ID="{1}" since the ' \
+                  'Tally does not contain this filter'.format(old_filter, self.id)
+            ValueError(msg)
+
+        self._filters.remove(old_filter)
+
+    def remove_nuclide(self, nuclide):
+        """Remove a nuclide from the tally
+
+        Parameters
+        ----------
+        nuclide : openmc.Nuclide
+            Nuclide to remove
+
+        """
+
+        if nuclide not in self.nuclides:
+            msg = 'Unable to remove nuclide "{0}" from Tally ID="{1}" since the ' \
+                  'Tally does not contain this nuclide'.format(nuclide, self.id)
+            ValueError(msg)
+
+        self._nuclides.remove(nuclide)
+
+    def _can_merge_filters(self, other):
+        """Determine if another tally's filters can be merged with this one's
+
+        The types of filters between the two tallies must match identically.
+        The bins in all of the filters must match identically, or be mergeable
+        in only one filter. This is a helper method for the can_merge(...)
+        and merge(...) methods.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for mergeable filters
+
+        """
+
+        # Two tallies must have the same number of filters
+        if len(self.filters) != len(other.filters):
+            return False
+
+        # Return False if only one tally has a delayed group filter
+        tally1_dg = self.contains_filter(openmc.DelayedGroupFilter)
+        tally2_dg = other.contains_filter(openmc.DelayedGroupFilter)
+        if sum([tally1_dg, tally2_dg]) == 1:
+            return False
+
+        # Look to see if all filters are the same, or one or more can be merged
+        for filter1 in self.filters:
+            merge_filters = False
```

Comment:
I'm not sure if I see the point of using `merge_filters`.
```
if filter1 == filter2 => start a new loop, and reset merge_filters/mergeable_filter

elif filter1.can_merge(filter2) and not merge_filters => start a new loop, and reset merge_filters/mergeable_filter

elif filter1.can_merge(filter2) and merge_filters => entire function returns false
```

`merge_filters` will always be false before hitting any of these conditions.

We should also rename this as just `mergeable`